test/fail_compilation/diag4596.d

```diff
@@ -1,10 +1,12 @@
 /*
 TEST_OUTPUT:
 ---
-fail_compilation/diag4596.d(15): Error: `this` is not an lvalue and cannot be modified
```

Comment:
The branches of the `?:` operator may return different things, e.g. an rvalue and an rvalue; this way the error is printed depending on the branch